### PR TITLE
Implement exporting from the command line

### PR DESCRIPTION
**Pull request by Huitsi**
This is a quick and basic implementation of exporting from the command line which would allow automatic "compilation" of SFXR files.

### CODE REVIEW

File: ui/main.cpp [hunk 1]
```diff
@@ -31,17 +32,45 @@ static void registerQmlTypes() {
...
 
 static void processArguments(QCommandLineParser* parser, QQmlApplicationEngine* engine) {
     parser->process(*qApp);
     const auto args = parser->positionalArguments();
+
     if (args.isEmpty()) {
         return;
     }
-    const QUrl url = QUrl::fromUserInput(args.first());
+    const QUrl url = QUrl::fromUserInput(args.first(),QDir::currentPath(),QUrl::AssumeLocalFile);
```
Comment (agateau) [inline ui/main.cpp]: style: can you add spaces after the `,`?

**Review by agateau — COMMENTED** [review]
Thanks a ton for this PR! I like the idea. I think it should be possible to use the WavSaver class directly without starting the QML engine. This would avoid the need to add setters in FileActions.qml. Do you think you can have a look at this approach? If not I can merge your PR and investigate this on my own.

PS: Do you use SFXR-Qt for some game or app development?

**Comment by Huitsi** [comment]
It looks to me a more direct access would require some refactoring and I admittedly don't know Qt or this codebase well enough to do that.

I'm afraid I've until now only used the original SFXR, and not too much of that either. My only public use of it is the two sounds in [Parallel Overhead](https://github.com/Huitsi/ParallelOverhead) which is also probably the first place I'd use this feature.

**Comment by agateau** [comment]
OK, no problem, will look into it when I get time. Thanks a lot! Merging it in.

**Comment by agateau** [comment]
I did the refactor. It changes one behavior: when using the `--export` option without `--output`, the name of the generated .wav  file is now `<sound>.wav`, not `<sound>.sfxr.wav`.

Oh, and Parallel Overhead is great! If a bit hypnotic :)

**Comment by Huitsi** [comment]
Great! I have meanwhile thought of a couple of things I'd like to tweak, so there may be another PR coming.

Most importantly I'd like your opinion on the actual option names since it's best to get them right early. Would something more consistent like --sample-bits and --sample-frequency be better? I also originally avoided -f since it's usually --force, but it could be used for the sampling frequency just as well.

(Thanks!)

**Comment by agateau** [comment]
Great to hear you are interested in doing more work on this topic!

I very much agree with you it is better to get option names right early and also that -f is usually for --force, so it's better not to use it. I had a look at [sox](http://sox.sourceforge.net/sox.html), a command-line tool to manipulate audio which has been around for a long time, for inspiration. It uses -b, --bits for the bits per sample, just like sfxr-qt and -r, --rate for the sample rate. This avoids the issue of using -f.

I think we should go this way. What do you think?

**Comment by Huitsi** [comment]
Consistency with an established program is good and--rate also avoids possible confusion with sampling vs audio frequency while not getting overly long. Of course, -r is also used for --recursive by many programs but conflicts are unavoidable with just 26 choices. I shall make the change with the other adjustments I've planned.

**Comment by agateau** [comment]
Sounds good!

**Pull request merged**